erpnext/selling/doctype/customer/customer.js

```diff
@@ -63,6 +63,12 @@ frappe.ui.form.on("Customer", {
 				}
 			}
 		});
+
+		frm.set_query("user", "portal_users", function() {
+			return {
+				query: "erpnext.selling.doctype.customer.customer.get_users_with_customer_role",
+			};
```

Comment:
Maybe custom query is not required at all. All you need is filters right? :grimacing: 


This filter will allow showing website users too (disabled by default) `filter: {"ignore_user_type": True}`

Reply:
reference: https://github.com/frappe/frappe/blob/4f70200bd5a259095b34e255ed664d58dfdea924/frappe/core/doctype/user/user.py#L961-L972